crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -125,6 +125,15 @@ def repocheck
       end
     end
 
+    def ha_deployed?
+      founders = NodeObject.find("pacemaker_founder:true AND pacemaker_config_environment:*")
+
+      if founders.empty?
+        Rails.logger.warn("No pacemaker cluster is configured.")
```

Comment:
It might be necessary now. But we definitely need to show the errors/warnings somehow to user, not just return true/false values from methods. 
I didn't change it here though, as we must do it correctly for all upgrade related methods.